bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/discovery/MiIoDiscovery.java

```diff
@@ -141,6 +204,12 @@ private void discovered(String ip, byte[] response) {
         String label = "Xiaomi Mi Device " + id + " (" + Long.parseUnsignedLong(id, 16) + ")";
         String country = "";
         boolean isOnline = false;
+        if (cloudDevices.containsKey(id)) {
+            if (cloudDevices.get(id).contains(ip)) {
+                logger.debug("Skipped adding local found {}. Already discovered by cloud.", label);
+                return;
+            }
+        }
```

Comment:
Is there a reason you did `contains` instead of `equals`? If not then you can simplify the code.
```suggestion
        if(ip.equals(cloudDevices.get(id)){
            logger.debug("Skipped adding local found {}. Already discovered by cloud.", label);
            return;
        }
```

Reply:
Yours is better indeed. The contains was only to avoid the NPE